src/script/components/Modals/CreateConversation/hooks/useCreateConversation.ts

```diff
@@ -0,0 +1,169 @@
+import {RECEIPT_MODE} from '@wireapp/api-client/lib/conversation/data';
+import {isNonFederatingBackendsError} from '@wireapp/core/lib/errors';
+import {amplify} from 'amplify';
+import {container} from 'tsyringe';
+
+import {WebAppEvents} from '@wireapp/webapp-events';
+
+import {Config} from 'src/script/Config';
+import {ACCESS_STATE} from 'src/script/conversation/AccessState';
+import {
+  toggleFeature,
+  teamPermissionsForAccessState,
+  ACCESS_TYPES,
+  ACCESS_MODES,
+} from 'src/script/conversation/ConversationAccessPermission';
+import {useSidebarStore, SidebarTabs} from 'src/script/page/LeftSidebar/panels/Conversations/useSidebarStore';
+import {RootContext} from 'src/script/page/RootProvider';
+import {generateConversationUrl} from 'src/script/router/routeGenerator';
+import {createNavigateKeyboard, createNavigate} from 'src/script/router/routerBindings';
+import {TeamState} from 'src/script/team/TeamState';
+import {useKoSubscribableChildren} from 'Util/ComponentUtil';
+import {isKeyboardEvent} from 'Util/KeyboardUtil';
+import {replaceLink, t} from 'Util/LocalizerUtil';
+
+import {PrimaryModal} from '../../PrimaryModal';
+import {useCreateConversationModal} from '../hooks/useCreateConversationModal';
+import {ConversationAccess, ConversationCreationStep, ConversationModerator, ConversationType} from '../types';
+
+export const useCreateConversation = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const {
+    conversationName,
+    hideModal,
+    showModal,
+    selectedContacts,
+    setConversationName,
+    setConversationCreationStep,
+    isReadReceiptsEnabled,
+    isServicesEnabled,
+    isGuestsEnabled,
+    conversationType,
+    access: conversationAccess,
+    moderator,
+  } = useCreateConversationModal();
+  const {setCurrentTab: setCurrentSidebarTab} = useSidebarStore();
+
+  const mainViewModel = useContext(RootContext);
+  const {content: contentViewModel} = mainViewModel!;
+  const {conversation: conversationRepository} = contentViewModel.repositories;
+  const teamState = container.resolve(TeamState);
+  const {isTeam, isMLSEnabled} = useKoSubscribableChildren(teamState, ['isTeam', 'isMLSEnabled']);
+
+  const defaultProtocol = isMLSEnabled
+    ? teamState.teamFeatures()?.mls?.config.defaultProtocol
+    : ConversationProtocol.PROTEUS;
+
+  const getAccessState = () => {
+    let access = ACCESS_STATE.TEAM.TEAM_ONLY;
+    if (isGuestsEnabled) {
+      access = toggleFeature(teamPermissionsForAccessState(ACCESS_STATE.TEAM.GUEST_FEATURES), access);
+    }
+
+    if (conversationAccess === ConversationAccess.Public && conversationType === ConversationType.Channel) {
+      access = toggleFeature(ACCESS_MODES.LINK, access);
+    }
+
+    if (isServicesEnabled && conversationType !== ConversationType.Channel) {
+      access = toggleFeature(ACCESS_TYPES.SERVICE, access);
+    }
+
+    return access;
+  };
```

Comment:
not a big deal, just a minor suggestion:

```suggestion
    if (isGuestsEnabled) {
      return toggleFeature(teamPermissionsForAccessState(ACCESS_STATE.TEAM.GUEST_FEATURES), access);
    }

    if (conversationAccess === ConversationAccess.Public && conversationType === ConversationType.Channel) {
      return toggleFeature(ACCESS_MODES.LINK, access);
    }

    if (isServicesEnabled && conversationType !== ConversationType.Channel) {
     return toggleFeature(ACCESS_TYPES.SERVICE, access);
    }
  };
```

Reply:
We can't return the result, `access` needs to be updated for each case.